src/AutorestSwift/Models/Complex/ConstantSchema.swift

```diff
@@ -76,10 +76,8 @@ class ConstantSchema: Schema {
 
     public required init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
-
-        valueType = try Schema.decode(withContainer: container, useKey: "valueType") ?? container
-            .decode(Schema.self, forKey: .valueType)
-        value = try container.decode(ConstantValue.self, forKey: .value)
+        valueType = try Schema.decode(withContainer: container, useKey: "valueType")!
```

Comment:
why we don't try to decode '
            container..decode(Schema.self, forKey: .valueType)'  when  Schema.decode failed? I think it is possible that we don't have all the possible Schema handled in Schema.decode.

Reply:
Because we already have that as the default case in `Schema.decode(withContainer:useKey:)`. We could put it in the catch clause as a fallback, but it doesn't see that this behavior was needed by anything in Autorest.

Reply:
I updated the catch clause of `Schema.decode` to try this instead of just automatically returning nil.